Yosemite/Yosemite/Stores/ProductVariationStore.swift

```diff
@@ -202,6 +204,25 @@ private extension ProductVariationStore {
             onCompletion(nil)
         }
     }
+
+    /// Deletes the product variation.
+    ///
+    func deleteProductVariation(productVariation: ProductVariation, onCompletion: @escaping (Result<ProductVariation, ProductUpdateError>) -> Void) {
+        remote.deleteProductVariation(siteID: productVariation.siteID,
+                                      productID: productVariation.productID,
+                                      variationID: productVariation.productVariationID) { [weak self] result in
+            guard let self = self else {
+                return
+            }
+            switch result {
+            case .failure(let error):
+                onCompletion(.failure(ProductUpdateError(error: error)))
+            case .success(let productVariation):
+                self.deleteStoredProductVariation(siteID: productVariation.siteID, productVariationID: productVariation.productVariationID)
+                onCompletion(.success(productVariation))
```

Comment:
nit: I think having the success case first and the failure case later it's a bit more aligned with the rest of the project!